course5/homework05/vector2d.h

```diff
@@ -0,0 +1,68 @@
+#ifndef vector2d_h
+#define vector2d_h
+
+#include <string>
+#include <cstdio>
+#include <iostream>
+#include <cmath>
+
+// ARS -- week 05. 20170301
+
+namespace vector2D
+{
+
+class Vector2D;
+
+class Vector2D
+{
+private:
+	double mX;
+	double mY;
+public:
+	Vector2D();
+	Vector2D(double x, double y);
+	~Vector2D();
+	double GetX() const;
+	double GetY() const;
+	void SetX(double x);
+	void SetY(double y);
+	double GetPolarR();
+	double GetPolarPhi();
+	Vector2D& operator+ ();  // unary, idempotent funtion
+	Vector2D operator+ (const Vector2D& rhs);
+	Vector2D operator- (); // unary
+	Vector2D operator- (const Vector2D& rhs);
+	Vector2D& operator+=(const Vector2D& rhs);
+	Vector2D& operator-=(const Vector2D& rhs);
+	bool operator==(const Vector2D& rhs);
+	bool operator!=(const Vector2D& rhs);
+
+	Vector2D operator*(const double r); // for polar init: multiply the (cos phi, sin phi) vector with the radius r
```

Comment:
Multiplying vector by scalar is useful operation in general, not only for initialisation.
You can make it even more useful by allowing for double*Vector2D